src/main/java/com/dreamhorizon/enchantments/listener/EntityListener.java

```diff
@@ -63,4 +60,42 @@ public void onPlayerAttackPlayer(EntityDamageByEntityEvent event) {
             }
         }
     }
+
+    @EventHandler
+    public void onEntityDeath(EntityDeathEvent event) {
+        LivingEntity entityThatDied = event.getEntity();
+
+        if (event.getEntity().getKiller() != null) {
+            LivingEntity killer = entityThatDied.getKiller();
+            if (killer != null) {
+
+                ItemStack attackItem = getAttackItem(killer);
+                if (attackItem == null) {
+                    return;
+                }
+
+                if (attackItem.getType() == Material.AIR
+                    || attackItem.getType() == Material.CAVE_AIR
+                    || attackItem.getType() == Material.VOID_AIR) {
+                    return;
+                }
+
+                if (attackItem.containsEnchantment(DHEnchantments.CANNIBALISM)) {
+                    Cannibalism.addCannibalismEffect(attackItem, entityThatDied, killer);
+                }
+
+                Cannibalism.addCannibalismEffect(attackItem, entityThatDied, killer);
```

Comment:
Are you sure that the second call to Cannibalism.addCanibalismEffect should be here?

Reply:
Good catch, thanks